test/e2e/apicoverage/apicoverage.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+var (
+	openAPIFile = flag.String("openapi", "", "File path to openapi-spec of Kubernetes. If not specifying, the openapi-spec is download from https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json instead")
+	restLog     = flag.String("restlog", "", "File path to REST API operation log of Kubernetes")
+)
+
+// Standard HTTP methods: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#path-item-object
+func isHTTPMethod(method string) bool {
+	methods := []string{"get", "put", "post", "delete", "options", "head", "patch"}
+	for _, validMethod := range methods {
+		if method == validMethod {
+			return true
+		}
+	}
+	return false
+}
+
+type apiData struct {
+	Method string
+	URL    string
+}
+
+type apiArray []apiData
+
+var reOpenapi = regexp.MustCompile(`({\S+})`)
+
+func parseOpenAPI(openapi string) apiArray {
+	var decodeData interface{}
```

Comment:
better to use [go-openapi/spec.Swagger](https://github.com/go-openapi/spec/blob/master/swagger.go#L30)?

Reply:
@mbohlool Oh, that is a nice idea. Yeah, it would be nice to avoid re-inventing the wheel.